crates/agglayer-contracts/src/lib.rs

```diff
@@ -182,11 +305,31 @@ where
 
         use crate::contracts::PolygonZkEvmGlobalExitRootV2::InitL1InfoRootMap;
 
+        // Find the deployment block of the rollup manager contract to optimize
+        // InitL1InfoRootMap event search.
+        let rollup_manager_deployment_block = find_contract_deployment_block_number(
+            &*rpc,
+            polygon_zkevm_global_exit_root_v2_contract,
+        )
+        .await
+        .inspect_err(|error| {
+            error!(
+                ?error,
+                "Failed to find contract {} deployment block",
+                polygon_zkevm_global_exit_root_v2_contract
+            );
+        })
+        .unwrap_or(None);
```

Comment:
Have you considered making this a hard error? Seems to me that if the contract address is not found, we won't be able to do much else anyway.